### PR TITLE
Extend LoadPolicy to load method beyond POST

### PR DESCRIPTION
**Pull request by egvili**
### What does this PR do?
Add parameter to the function Policy::LoadPolicy() to allow a different load method other than POST. POST is still the default value
Additionally, modify build to use NuGet to pack and upload artifact to Conjur artifactory

### What ticket does this PR close?
Resolves #[https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-10261]

#### Change log
- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
```
Comment (aloncarmel111) [inline CHANGELOG.md:8]: added

File: Jenkinsfile [hunk 2]
```diff
@@ -35,7 +35,10 @@ pipeline {
 
     stage('Build and test package') {
       steps {
-        sh './build.sh'
+        script {          
+          ARTIFACT_REMOTE_DIRECTORY = "${env.BUILD_NUMBER}.0.0-${env.BRANCH_NAME.replace('/','-')}"
```
Comment (aloncarmel111) [inline Jenkinsfile:39]: rename ARTIFACT_REMOTE_DIRECTORY should be version number no?

File: docker/Makefile [hunk 3]
```diff
@@ -10,7 +10,7 @@ TAG := ${REPO}:$(shell date -u +%Y%m%dT%H%M%SZ)
 tag: $(FILES)
 	docker build -t ${TAG} .
 	echo ${TAG} > tag
-	docker tag -f ${TAG} ${REPO}
+	docker tag ${TAG} ${REPO}
```
Comment (aloncarmel111) [inline docker/Makefile:13]: why -f is not needed

File: CHANGELOG.md (hunk 1 above)
Comment (egvili) [reply CHANGELOG.md:8]: Added although we change an existing function?
When do we use Changed then?

File: Jenkinsfile (hunk 2 above)
Comment (egvili) [reply Jenkinsfile:39]: Correct. That's a more reasonable name. I'll change

File: docker/Makefile (hunk 3 above)
Comment (egvili) [reply docker/Makefile:13]: The flag -f is not supported anymore by docker.
It seems to be doing the tagging just fine even without the -f flag

File: CHANGELOG.md (hunk 1 above)
Comment (egvili) [reply CHANGELOG.md]: I changed to Added per request

**Review by aloncarmel111 — DISMISSED** [review]
LGTM

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Add parameter to the function `Policy::LoadPolicy()` to allow a different load method other than POST. POST is still the default value
```
Comment (nahumcohen) [inline CHANGELOG.md:9]: Should add the supported methods in Conjur

File: conjur-api/conjur-api.nuspec [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <authors>CyberArk Software Ltd</authors>
+    <owners>CyberArk Software Ltd</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>conjur-api-dotnet</description>
+    <copyright>CyberArk Copyright 2020</copyright>
+  </metadata>
+  <files>
+    <file src="bin/Release/conjur-api.dll" target="lib/net452" />
+  </files>
+</package>
```
Comment (nahumcohen) [inline conjur-api/conjur-api.nuspec:16]: add new line

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
 
+RUN apt-get update -y && apt-get install -y gnupg2
+
 # This is for osslsigncode
 ADD dividedmind.pub /tmp
 RUN apt-key add - < /tmp/dividedmind.pub && rm /tmp/dividedmind.pub
 RUN echo 'deb http://download.opensuse.org/repositories/home:/dividedmind:/conjur/Debian_7.0/ /' >> /etc/apt/sources.list.d/dividedmind.list
 
-RUN apt-get update -y
 RUN apt-get install -y nunit-console git osslsigncode
```
Comment (nahumcohen) [inline docker/Dockerfile:10]: Does nunit-console require here?

**Review by nahumcohen — APPROVED** [review]

**Pull request merged**